### PR TITLE
Measurement of Method protection against nullability of arguments.

### PR DESCRIPTION
**Pull request by nimakarimipour**
This PR is a followup to enable nullability inference implemented in [AutoFixer](https://github.com/nimakarimipour/NullAwayAutoFixer). This PR provides the facility to measure protection of methods against nullability of arguments. When activated, it transforms all method arguments's state at certain index to `@Nullable`.

It adds the following configurations settings in `FixSerializationConfig`:
* `paramTestIndex`: Index of method argument
* `methodParamProtectionTestEnabled`: if enabled, method parameters at index `paramTestIndex` in all methods, will be at `@Nullable` state at start point of the method.

Please note, `SerializeFixMetadata` flag must be set to `true` to enable the service above.

Please see the example below where `NullAway` is not reporting any error:

```java
class C{
      Object run(Object foo, Object bar) {
            return bar;
      }
}
```

However, if `methodParamProtectionTestEnabled` is set to `true` and `paramTestIndex` is set to one, `bar` which is at index 1 of arguments list will be at `@Nullable` state at the start point of the method and will result to an error:

```java
class C{
      Object run(Object foo, Object bar) {
            return bar; [NullAway]: error: returning @Nullable from @Nonnull method.
      }
}
```

To activate the feature above entry below must be present in the serialization config file:
```xml
<serialization>
    <paramTest activation="true", index=0/>
</serialization>
```

With best regards

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java [hunk 1]
```diff
@@ -51,6 +51,19 @@
    * computed at request.
    */
   public final boolean suggestEnclosing;
+
+  /**
+   * If enabled, method parameters at index {@link FixSerializationConfig#paramTestIndex} in all
+   * methods, will be at {@code @Nullable} state at start point of the method.
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java:57]:
```suggestion
   * If enabled, the formal parameter at index {@link FixSerializationConfig#paramTestIndex} in all
   * methods will be treated as {@code @Nullable}
```

File: nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java [hunk 2]
```diff
@@ -51,6 +51,19 @@
...
+   */
+  public final boolean methodParamProtectionTestEnabled;
+
+  /**
+   * Index of the parameter at methods argument list which will be considered {@code @Nullable}, if
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java:62]:
```suggestion
   * Index of the formal parameter of all methods which will be considered {@code @Nullable}, if
```

File: nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java [hunk 3]
```diff
@@ -110,6 +129,15 @@ public FixSerializationConfig(String configFilePath) {
       throw new IllegalStateException(
           "Error in the fix serialization configuration, suggest flag must be enabled to activate enclosing method and class serialization.");
     }
+    methodParamProtectionTestEnabled =
+        XMLUtil.getValueFromAttribute(document, "/serialization/paramTest", "active", Boolean.class)
+            .orElse(false);
+    paramTestIndex =
+        (int)
+            Math.round(
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java:137]: Does XML somehow not support integers, so we have to do this rounding?

**Review by msridhar — CHANGES_REQUESTED** [review]
LGTM!  Just a couple of minor comments and suggestions

File: nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java (hunk 3 above)
Comment (nimakarimipour) [reply nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java]: Yes it does, actually I am only working with strings and it was my mistake from converting `json` implementation to `xml` and this was left from there. Thank you for noting it. Resolved it now. 1c9e0bfb3dd002cade3e0b372bf968eb615f0232

**Comment by nimakarimipour** [comment]
Hi @msridhar, I believe this is ready for another round of review.

File: nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java [hunk 4]
```diff
@@ -110,6 +129,12 @@ public FixSerializationConfig(String configFilePath) {
       throw new IllegalStateException(
           "Error in the fix serialization configuration, suggest flag must be enabled to activate enclosing method and class serialization.");
     }
+    methodParamProtectionTestEnabled =
+        XMLUtil.getValueFromAttribute(document, "/serialization/paramTest", "active", Boolean.class)
+            .orElse(false);
+    paramTestIndex =
+        XMLUtil.getValueFromAttribute(document, "/serialization/paramTest", "index", Integer.class)
+            .orElse(1000);
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java:137]: The constant 1000 is a bit weird here, let's be consistent:
```suggestion
            .orElse(Integer.MAX_VALUE);
```

**Review by msridhar — APPROVED** [review]
LGTM, one minor comment that I will fix

**Pull request merged**